Testscripts/Linux/SetupHbKernel.sh

```diff
@@ -127,13 +123,23 @@ function Main() {
 		cd
 
 		# Append the test log to the main log files.
-		cat /usr/src/linux/TestExecution.log >> TestExecution.log
-		cat /usr/src/linux/TestExecutionError.log >> TestExecutionError.log
+		if [ -f /usr/src/linux/TestExecution.log ]; then
+			cat /usr/src/linux/TestExecution.log >> TestExecution.log
+		fi
+		if [ -f /usr/src/linux/TestExecutionError.log ]; then
+			cat /usr/src/linux/TestExecutionError.log >> TestExecutionError.log
+		fi
 	fi
 
 	if [[ "$DISTRO" =~ "redhat" ]];then
-		sed -i -e "s/rootdelay=300/rootdelay=300 resume=$sw_uuid/g" /etc/default/grub
-		LogMsg "$?: Updated the /etc/default/grub with resume=$sw_uuid"
+		_entry=$(cat /etc/default/grub | grep 'rootdelay=')
+		if [ $_entry ]; then
+			sed -i -e "s/rootdelay=300/rootdelay=300 resume=$sw_uuid/g" /etc/default/grub
+			LogMsg "$?: Updated the /etc/default/grub with resume=$sw_uuid"
+		else
+			echo GRUB_CMDLINE_LINUX_DEFAULT="console=tty1 console=ttyS0 earlyprintk=ttyS0 rootdelay=300 resume=$sw_uuid" >> /etc/default/grub
+			LogMsg "$?: Added resume=$sw_uuid in /etc/default/grub file"
+		fi
```

Comment:
when GRUB_CMDLINE_LINUX_DEFAULT exist but without rootdelay=300? I am not sure what happened if there are two lines for GRUB_CMDLINE_LINUX_DEFAULT, I think we can find line of GRUB_CMDLINE_LINUX_DEFAULT and add resume=$sw_uuid in the end.

Reply:
"when GRUB_CMDLINE_LINUX_DEFAULT exist but without rootdelay=300?": that was what happened in the recent change in Ubuntu system, so I tested with @dcui. 
"if there are two lines for GRUB_CMDLINE_LINUX_DEFAULT": are you talking about a bug in Canonical? I am not sure what it meant to us of testing on buggy system. It's invalid in the beginning. @dcui gave me the line 163. 
Added rootdelay=300 in the line 163. 4a6e757ce87a2d7fb5572dcf3349430a16d32909